_posts/2017-10-17-faf-in-openshift.markdown

```diff
@@ -0,0 +1,160 @@
+<br><br>
+To get IP address of the pod:
+```
+{% raw %}oc get pod <pod_name> --template='PodIP: {{.status.podIP}}{{"\n"}}'{% endraw %}
+```
+
+
+## Web Interface
+
+It is also possible to deploy FAF through the web interface, except for editing security context
+constraints. You will need to use the command line for it.
+
+1. Create a new project.
+[![create_project][create_project]][create_project]
+
+2. Add FAF's docker image to project. Use image name ```abrt/faf-image/```.
+[![add_docker_image][add_docker_image]][add_docker_image]
+
+3. Create a new service account and add it to **anyuid's** SCC.
+```
+oc create serviceaccount <sa_username>
+oc adm policy add-scc-to-user anyuid -z <sa_username>
+oc patch dc/<app_name> --patch '{"spec":{"template":{"spec":{"serviceAccountName": "<sa_username>"}}}}'
+```
+The FAF should now be running on ```http://<pod_ip>/faf```.
+[![success_deploy][success_deploy]][success_deploy]
+
+
+# Persistent storage for FAF
+
+Containers are not persistent by default; their contents get cleared, when you restart them.
+Default claimed volume for the pod is EmptyDir, which is a temporary directory that will get cleared, once the pod is terminated.
+Again, you have an option to do this either using the command line or the web interface.
+
+### Command line
+
+Create a volume claim for database:
+```yaml
+apiVersion: "v1"
+kind: "PersistentVolumeClaim"
+metadata:
+  name: "faf-database"
+spec:
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "5Gi"
+```
+```
+oc create -f db-pvc-claim.yaml
+```
+
+Create a volume claim for FAF's data:
+```yaml
+apiVersion: "v1"
+kind: "PersistentVolumeClaim"
+metadata:
+  name: "faf-report-data"
+spec:
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "5Gi"
+```
+```
+oc create -f report-pvc-claim.yaml
+```
+
+Update the volume configuration:
+```
+oc volume dc/<app_name> --add --overwrite --claim-name=faf-database --name=faf-1 -t=persistentVolumeClaim -m=/var/lib/postgres
+
+oc volume dc/<app_name> --add --overwrite --claim-name=faf-report-data --name=faf-2 -t=persistentVolumeClaim -m=/var/spool/faf
+```
+
+### Web interface
+1. Create claims for storage. One for database and another for FAF's data.
+<br>
+```Storage``` -> ```Create Storage```
+[![create_storage][create_storage]][create_storage]
+
+2. Remove mounted volumes.
+<br>
+```Applications``` -> ```Deployments```. Then choose the FAF deployment and switch to ```Configuration``` tab.
+[![container_volumes][container_volumes]][container_volumes]
+
+3. Add a new storage for database and FAF's data
+[![db_storage][db_storage]][db_storage]
+The mount path for database is ```/var/lib/postgres```, and ```/var/spool/faf``` for FAF's data.
+
+
+# Reporting into deployed FAF
+
+To be able to report to your newly created FAF instance, you will have to change uReport
+configuration in ```/etc/libreport/plugins/ureport.conf```.
```

Comment:
... on every machine from which you want gather crash reports. You can utilize Ansible role as stated here https://abrt.github.io/abrt/ansible/faf/url/setup/2017/10/15/ABRT-ansible-faf-url-setup/